command/agent/cache/lease_cache.go

```diff
@@ -0,0 +1,762 @@
+			Ctx:        baseCtx,
+			CancelFunc: baseCancel,
+		}
+
+		// Reset the memdb instance
+		if err := c.db.Flush(); err != nil {
+			return err
+		}
+
+	default:
+		return errInvalidType
+	}
+
+	c.logger.Debug("successfully cleared matching cache entries")
+
+	return nil
+}
+
+// handleRevocationRequest checks whether the originating request is a
+// revocation request, and if so perform applicable cache cleanups.
+// Returns true is this is a revocation request.
+func (c *LeaseCache) handleRevocationRequest(ctx context.Context, req *SendRequest, resp *SendResponse) (bool, error) {
+	// Lease and token revocations return 204's on success. Fast-path if that's
+	// not the case.
+	if resp.Response.StatusCode != http.StatusNoContent {
+		return false, nil
+	}
+
+	namespace, path := deriveNamespaceAndRevocationPath(req)
+
+	switch {
+	case path == vaultPathTokenRevoke:
+		// Get the token from the request body
+		jsonBody := map[string]interface{}{}
+		if err := json.Unmarshal(req.RequestBody, &jsonBody); err != nil {
+			return false, err
+		}
+		tokenRaw, ok := jsonBody["token"]
+		if !ok {
+			return false, fmt.Errorf("failed to get token from request body")
+		}
+		token, ok := tokenRaw.(string)
+		if !ok {
+			return false, fmt.Errorf("expected token in the request body to be string")
+		}
+
+		// Clear the cache entry associated with the token and all the other
+		// entries belonging to the leases derived from this token.
+		if err := c.handleCacheClear(ctx, "token", token); err != nil {
+			return false, err
+		}
+
+	case path == vaultPathTokenRevokeSelf:
+		// Clear the cache entry associated with the token and all the other
+		// entries belonging to the leases derived from this token.
+		if err := c.handleCacheClear(ctx, "token", req.Token); err != nil {
+			return false, err
+		}
+
+	case path == vaultPathTokenRevokeAccessor:
+		jsonBody := map[string]interface{}{}
+		if err := json.Unmarshal(req.RequestBody, &jsonBody); err != nil {
+			return false, err
+		}
+		accessorRaw, ok := jsonBody["accessor"]
+		if !ok {
+			return false, fmt.Errorf("failed to get accessor from request body")
+		}
+		accessor, ok := accessorRaw.(string)
+		if !ok {
+			return false, fmt.Errorf("expected accessor in the request body to be string")
+		}
+
+		if err := c.handleCacheClear(ctx, "token_accessor", accessor); err != nil {
+			return false, err
+		}
+
+	case path == vaultPathTokenRevokeOrphan:
+		jsonBody := map[string]interface{}{}
+		if err := json.Unmarshal(req.RequestBody, &jsonBody); err != nil {
+			return false, err
+		}
+		tokenRaw, ok := jsonBody["token"]
+		if !ok {
+			return false, fmt.Errorf("failed to get token from request body")
+		}
+		token, ok := tokenRaw.(string)
+		if !ok {
+			return false, fmt.Errorf("expected token in the request body to be string")
+		}
+
+		// Find out all the indexes that are directly tied to the revoked token
+		indexes, err := c.db.GetByPrefix(cachememdb.IndexNameToken.String(), token)
```

Comment:
Why is this (and the call below) GetByPrefix instead of simply Get?

Reply:
The goal of this call is to get all the entries in the cache that are attached to the token. While `GetByPrefix` returns all the attached entries, `Get` returns the first match.

Reply:
Got it, thanks.